GameOfLifeApp/GameLogic.cs

```diff
@@ -0,0 +1,153 @@
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace GameOfLifeApp
+{
+    public class GameLogic
+    {
+        /// <summary>
+        /// Holds user data of array size.    
+        /// </summary>        
+        public int CountOfRows { get; set; }
+
+        /// <summary>
+        /// Holds user data of array size. 
+        /// </summary>        
+        public int CountOfColumns { get; set; }
+
+        /// <summary>
+        /// Currently active game field.
+        /// </summary>        
+        public bool[,] CurrentField { get; set; }
+
+        /// <summary>
+        /// Defines array size.
+        /// </summary>        
+        public GameLogic(int rows, int columns)
+        {
+            CurrentField = new bool[rows, columns];
+            CountOfRows = rows;
+            CountOfColumns = columns;
+        }
+
+        /// <summary>
+        /// Generates a start array of random values according to user`s input.
+        /// </summary>
+        public void SetUpField()
+        {
+            Random rnd = new Random();
+
+            bool[,] GeneratedField = new bool[CountOfRows, CountOfColumns];
+
+            for (int row = 0; row < CountOfRows; row++)
+            {
+                for (int column = 0; column < CountOfColumns; column++)
+                {
+                    GeneratedField[row, column] = rnd.Next(0, 2) == 0;
+                }
+            }
+            CurrentField = GeneratedField;
+        }
+
+        /// <summary>
+        /// Populates the field with Glider the Shape. 
+        /// </summary>
+        public void CreateGliderShape()
+        {
+            CurrentField[0, 0] = true;
+            CurrentField[2, 0] = true;
+            CurrentField[1, 1] = true;
+            CurrentField[2, 1] = true;
+            CurrentField[1, 2] = true;
+        }
+
+        /// <summary>
+        /// Gets number of alive cells around a given cell. 
+        /// </summary>
+        /// <param name="currentRow">Row of a current cell.</param>
+        /// <param name="currentColumn">Column of a current cell.</param>
+        /// <returns>Returns number of alive neighbours.</returns>
+        private int NeighboursCount(int currentRow, int currentColumn)
+        {
+            int count = 0;
+            for (int row = currentRow - 1; row <= currentRow + 1; row++)
+            {
+                for (int column = currentColumn - 1; column <= currentColumn + 1; column++)
+                {
+                    int actualRow;
+                    int actualColumn;
+                    if (row < 0)
+                    {
+                        actualRow = CurrentField.GetLength(0) - 1;
+                    }
+                    else if (row == CurrentField.GetLength(0))
+                    {
+                        actualRow = 0;
+                    }
+                    else
+                    {
+                        actualRow = row;
+                    }
```

Comment:
the actual row can be set in the row loop as it won't change when going to a different column.